views_api.py

```diff
@@ -29,14 +32,159 @@
     CreateTposData,
     CreateTposInvoice,
     CreateUpdateItemData,
+    InventorySale,
     PayLnurlWData,
     TapToPay,
     Tpos,
 )
 
+get_user_inventories: Callable[[str], Awaitable[Any]] | None = None
+inventory_db: Database | None = None
+inventory_get_items_by_ids: Callable[[str, list[str]], Awaitable[list[Any]]] | None = (
+    None
+)
+inventory_update_item: Callable[[Any], Awaitable[Any]] | None = None
+CreateInventoryUpdateLog: type | None = None
+InventoryItem: type | None = None
+UpdateSource: type | None = None
+
+if not TYPE_CHECKING:
+    try:  # inventory extension is optional
+        from ..inventory.crud import (
+            db as inventory_db,
+        )  # pyright: ignore[reportMissingImports]
+        from ..inventory.crud import (  # pyright: ignore[reportMissingImports]
+            get_inventories as get_user_inventories,
+        )
+        from ..inventory.models import (
+            Item as InventoryItem,
+        )
+    except Exception:  # pragma: no cover - guard when inventory extension is absent
+        pass
+
 tpos_api_router = APIRouter()
 
 
+def _inventory_tags_to_list(raw_tags: str | list[str] | None) -> list[str]:
+    if raw_tags is None:
+        return []
+    if isinstance(raw_tags, list):
+        return [tag for tag in raw_tags if tag]
+    return [tag for tag in raw_tags.split(",") if tag]
+
+
+def _inventory_tags_to_string(raw_tags: str | list[str] | None) -> str | None:
+    if raw_tags is None:
+        return None
+    if isinstance(raw_tags, str):
+        return raw_tags
+    return ",".join([tag for tag in raw_tags if tag])
+
+
+def _first_image(images: str | list[str] | None) -> str | None:
+    if not images:
+        return None
+    if isinstance(images, list):
+        return _normalize_image(images[0]) if images else None
+    raw = str(images).strip()
+    if not raw:
+        return None
+    try:
+        parsed = json.loads(raw)
+        if isinstance(parsed, list) and parsed:
+            return _normalize_image(parsed[0])
+    except Exception:
+        pass
+    if "|||" in raw:
+        return _normalize_image(raw.split("|||")[0])
+    if "," in raw:
+        return _normalize_image(raw.split(",")[0])
+    return _normalize_image(raw)
+
+
+def _normalize_image(val: str | None) -> str | None:
+    if not val:
+        return None
+    val = str(val).strip()
+    if not val:
+        return None
+    if val.startswith("http") or val.startswith("/api/") or val.startswith("data:"):
+        return val
+    return f"/api/v1/assets/{val}/binary"
+
+
+async def _get_default_inventory_id(user_id: str) -> str | None:
+    if get_user_inventories is None:
+        return None
+    inventory = await get_user_inventories(user_id)
+    return inventory.id if inventory else None
+
+
+async def _get_inventory_tags(inventory_id: str) -> list[str]:
+    if inventory_db is None:
+        return []
+    rows: list[Any] = await inventory_db.fetchall(
```

Comment:
I think we should avoid this DB interactions here!

Maybe add API endpoints inventory side?